### PR TITLE
 server: reduce the lock overhead

### PR DESCRIPTION
**Pull request by nolouch**
When the number of regions is large, the lock is  expensive when processing the heartbeat.now does not need to get the lock if cache not needs to update.

### CODE REVIEW

File: server/schedule/basic_cluster.go [hunk 1]
```diff
@@ -200,14 +200,14 @@ func (bc *BasicCluster) PutRegion(region *core.RegionInfo) error {
 	return nil
 }
 
-// UpdateWriteStatus update the write status
-func (bc *BasicCluster) UpdateWriteStatus(region *core.RegionInfo) {
+// IsUpdateWriteStatus update the write status
+func (bc *BasicCluster) IsUpdateWriteStatus(region *core.RegionInfo) (bool, uint64, *core.RegionStat) {
```
Comment (disksing) [inline server/schedule/basic_cluster.go]: I suppose the second return value is always the region's id, we can load it from region info, so don't need to return it.

File: server/schedule/basic_cluster.go [hunk 2]
```diff
@@ -249,28 +249,27 @@ func (bc *BasicCluster) UpdateWriteStatCache(region *core.RegionInfo, hotRegionT
...
 		newItem.FlowBytes = v.FlowBytes
 	}
-	bc.WriteStatistics.Put(key, newItem)
+	return true, key, newItem
 }
 
-// UpdateReadStatus update the read status
-func (bc *BasicCluster) UpdateReadStatus(region *core.RegionInfo) {
+// IsUpdateReadStatus update the read status
+func (bc *BasicCluster) IsUpdateReadStatus(region *core.RegionInfo) (bool, uint64, *core.RegionStat) {
```
Comment (disksing) [inline server/schedule/basic_cluster.go]: How about `checkReadStatus`?

**Comment by siddontang** [comment]
Please add a PR description.

File: server/schedule/basic_cluster.go [hunk 3]
```diff
@@ -200,14 +200,14 @@ func (bc *BasicCluster) PutRegion(region *core.RegionInfo) error {
 	return nil
 }
 
-// UpdateWriteStatus update the write status
-func (bc *BasicCluster) UpdateWriteStatus(region *core.RegionInfo) {
+// CheckWriteStatus checks the write status, return whether need update statistics and item
```
Comment (disksing) [inline server/schedule/basic_cluster.go]: return -> returns

**Comment by nolouch** [comment]
PTAL @Connor1996 @disksing

**Pull request merged**